File: 1_Arrays/1_MajorityElement.jl
# Arrays: Majority Element 
function MajorityElement(nums::Vector{Int})::Int
    majorityElementValue = nums[1]
    count = 0

    for n in nums
        if (count == 0)
            majorityElementValue = n
        end
        
        if (majorityElementValue == n)
            count += 1
        else
            count -= 1
        end
    end
    
    return majorityElementValue;
end 


function main()
    nums = [ 2, 9, 1, 1, 1, 2, 2, 2, 2, 2]
    majorityElementValue = MajorityElement(nums)
    println("Majority Element", majorityElementValue)
end 

main()

File: 1_Arrays/1_MajorityElement.cpp
// Array: Majority Element 
class Solution {
public:
    int majorityElement(vector<int>& arr) 
    {
        int majorityElement = arr[0];
        int count = 0;
        for(int i = 0; i<arr.size(); i++)
        {
            if(count == 0)
            {               
                majorityElement = arr[i];                
            }
            count += (majorityElement == arr[i]) ? 1 : -1;
        }
        return majorityElement;
    } 
};